server/metadata/tenant_test.go

```diff
@@ -0,0 +1,216 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"context"
+	"testing"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/stretchr/testify/require"
+	api "github.com/tigrisdata/tigrisdb/api/server/v1"
+	"github.com/tigrisdata/tigrisdb/schema"
+	"github.com/tigrisdata/tigrisdb/server/config"
+	"github.com/tigrisdata/tigrisdb/server/metadata/encoding"
+	"github.com/tigrisdata/tigrisdb/server/transaction"
+	"github.com/tigrisdata/tigrisdb/store/kv"
+)
+
+func TestTenantManager_CreateTenant(t *testing.T) {
+	fdbCfg, err := config.GetTestFDBConfig("../../..")
+	require.NoError(t, err)
+
+	kv, err := kv.NewFoundationDB(fdbCfg)
+	require.NoError(t, err)
+
+	tm := transaction.NewManager(kv)
+	t.Run("create_tenant", func(t *testing.T) {
+		ctx := context.TODO()
+		_ = kv.DropTable(ctx, encoding.ReservedSubspaceKey)
+
+		m := NewTenantManager()
+		tx, err := tm.StartTxWithoutTracking(context.TODO())
+		require.NoError(t, err)
+
+		require.NoError(t, m.CreateTenant(context.TODO(), tx, &TenantNamespace{"ns-test1", 2}))
+
+		tenant := m.GetTenant("ns-test1")
+		require.Equal(t, "ns-test1", tenant.namespace.Name())
+		require.Equal(t, uint32(2), tenant.namespace.Id())
+		require.Empty(t, tenant.databases)
+		require.NoError(t, tx.Commit(context.TODO()))
+		_ = kv.DropTable(ctx, encoding.ReservedSubspaceKey)
+	})
+	t.Run("create_multiple_tenants", func(t *testing.T) {
+		ctx := context.TODO()
+		_ = kv.DropTable(ctx, encoding.ReservedSubspaceKey)
+
+		m := NewTenantManager()
+		tx, err := tm.StartTxWithoutTracking(context.TODO())
+		require.NoError(t, err)
+
+		require.NoError(t, m.CreateTenant(context.TODO(), tx, &TenantNamespace{"ns-test1", 2}))
+		require.NoError(t, m.CreateTenant(context.TODO(), tx, &TenantNamespace{"ns-test2", 3}))
+
+		tenant := m.GetTenant("ns-test1")
+		require.Equal(t, "ns-test1", tenant.namespace.Name())
+		require.Equal(t, uint32(2), tenant.namespace.Id())
+		require.Empty(t, tenant.databases)
+
+		tenant = m.GetTenant("ns-test2")
+		require.Equal(t, "ns-test2", tenant.namespace.Name())
+		require.Equal(t, uint32(3), tenant.namespace.Id())
+		require.Empty(t, tenant.databases)
+
+		require.NoError(t, tx.Commit(context.TODO()))
+
+		_ = kv.DropTable(ctx, encoding.ReservedSubspaceKey)
+	})
+	t.Run("create_duplicate_tenant_error", func(t *testing.T) {
+		ctx := context.TODO()
+		_ = kv.DropTable(ctx, encoding.ReservedSubspaceKey)
+
+		m := NewTenantManager()
+		tx, err := tm.StartTxWithoutTracking(context.TODO())
```

Comment:
nit: you can use ctx instead of context.TODO() everywhere in the tests.

Reply:
yeah, missed it.